File: ListViewTest/app/src/main/java/com/example/listviewtest/MainActivity.java
package com.example.listviewtest;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;
import android.view.View;
import android.widget.AbsListView;
import android.widget.Adapter;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ListView;
import android.widget.TextView;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    Button btnAdd;
    Button btnModify;
    Button btnDel;

    TextView textView;
    ListView listView;

    int select = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btnAdd = findViewById(R.id.btnAdd);
        btnModify = findViewById(R.id.btnModi);
        btnDel = findViewById(R.id.btnDel);
        textView = findViewById(R.id.textView);
        listView = findViewById(R.id.listView);
        final ArrayList<String> data = getData();


        final ArrayAdapter adapter = new ArrayAdapter<String>(getApplicationContext(), android.R.layout.simple_list_item_single_choice, data);
        listView.setChoiceMode(AbsListView.CHOICE_MODE_SINGLE);
        listView.setAdapter(adapter);

        textView.setText(data.get(0));

        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                textView.setText(data.get(position));
                select = position;
            }
        });
        btnAdd.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                data.add("리스트 아이템"+(data.size()+1));
                adapter.notifyDataSetChanged();
            }
        });
        btnDel.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                data.remove(select);
                adapter.notifyDataSetChanged();

            }
        });
        final Context context = this;
        btnModify.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                final EditText editText = new EditText(getApplicationContext());

                new AlertDialog.Builder(context)
                        .setTitle("리스트 아이템 수정")
                        .setMessage("현재 데이터 : " + data.get(select))

                        .setIcon(R.mipmap.ic_launcher_round)
                        .setView(editText)
                        .setCancelable(false)
                        .setNegativeButton("취소", new DialogInterface.OnClickListener() {
                            @Override
                            public void onClick(DialogInterface dialog, int which) {

                            }
                        })
                        .setPositiveButton("수정", new DialogInterface.OnClickListener() {
                            @Override
                            public void onClick(DialogInterface dialog, int which) {
                                data.set(select, editText.getText().toString());
                                adapter.notifyDataSetChanged();
                            }
                        })
                        .show();
            }
        });
    }

    private ArrayList<String> getData() {
        ArrayList<String> arr = new ArrayList<>();
        for (int i = 0; i < 5; i++) {
            arr.add("리스트 데이터 " + (i + 1));
        }
        return arr;
    }
}

File: Chapter11/ex03_listview_addmodifydelete/src/main/java/com/example/ex03_listview_addmodifydelete/MainActivity.kt
package com.example.ex03_listview_addmodifydelete

import android.content.DialogInterface
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.ArrayAdapter
import android.widget.EditText
import android.widget.ListView
import androidx.appcompat.app.AlertDialog
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    var dataArr = getStringArray()

    var arrList = getArrayListData()


    private fun getStringArray(): Array<String>
    {

        return Array(100){ i->i.toString()+"hi"}
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        var adapter = ArrayAdapter<String>(this, android.R.layout.simple_list_item_1,arrList  )



        // lastView ArrayAdapter connect

        listView.adapter = adapter

        listView.setOnItemClickListener{
            parent, view, position, id ->
            textView.setText(arrList.get(position))
        }

        btnAdd.setOnClickListener {
            arrList.add("get"+(arrList.size).toString())
            adapter.notifyDataSetChanged()
        }
        btnModify.setOnClickListener{
            val count = adapter.count
            val checkedItemPostition = listView.checkedItemPosition
            val editText = EditText(applicationContext)
            AlertDialog.Builder(applicationContext)
                    .setTitle("listView Item Edit")
                    .setMessage("select data : " + arrList.get(checkedItemPostition))
                    .setIcon(R.mipmap.ic_launcher)
                    .setCancelable(false)
                    .setView(editText)
                    .setPositiveButton("Edit") { dialog, which -> arrList.set(checkedItemPostition,editText.text.toString())
                    }
                    .setNegativeButton("cancle",null)
                    .show()

            adapter.notifyDataSetChanged()
        }

        btnDelete.setOnClickListener{
            arrList.removeAt(arrList.size-1)
            adapter.notifyDataSetChanged()
        }


    }

    private fun getArrayListData(): ArrayList<String>{
        val a = ArrayList<String>()
        for (i in 0..5)
        {
            a.add("list item" + (i+1))
        }
        return a
    }


//    override fun onClick(v: View?) {
//        val count = adapter.count
//        val checkedItemPostition = listView.checkedItemPosition
//
//        if (v != null) {
//            when(v.id){
//                R.id.btnAdd -> {
//
//                }
//
//                R.id.btnModify -> {
//
//                    val editText = EditText(applicationContext)
//                    AlertDialog.Builder(applicationContext)
//                            .setTitle("listView Item Edit")
//                            .setMessage("select data : " + arrList.get(checkedItemPostition))
//                            .setIcon(R.mipmap.ic_launcher)
//                            .setCancelable(false)
//                            .setView(editText)
//                            .setPositiveButton("Edit") { dialog, which -> arrList.set(checkedItemPostition,editText.text.toString())
//                            }
//                            .setNegativeButton("cancle",null)
//                            .show()
//
//                }
//                R.id.btnDelete -> {
//                    arrList.removeAt(checkedItemPostition)
//                }
//            }
//            adapter.notifyDataSetChanged()
//        }
//    }
}
